presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -587,6 +595,112 @@ private void testPredicating(Session session, FileFormat fileFormat)
         dropTable(session, "test_predicating_on_real");
     }
 
+    @Test
+    public void testMetadataDeleteSimple()
+    {
+        testWithAllFileFormats(this::testMetadataDeleteSimpleForFormat);
+    }
+
+    private void testMetadataDeleteSimpleForFormat(Session session, FileFormat format)
+    {
+        assertUpdate(session, format("CREATE TABLE test_metadata_delete_simple (col1 BIGINT, col2 BIGINT) WITH (format = '%s', partitioning = ARRAY['col1'])", format.name()));
+        assertUpdate(session, "INSERT INTO test_metadata_delete_simple VALUES(1, 100), (1, 101), (1, 102), (2, 200), (2, 201), (3, 300)", 6);
+        assertQueryFails(
+                session,
+                "DELETE FROM test_metadata_delete_simple WHERE col1 = 1 AND col2 > 101",
+                "This connector only supports delete where one or more partitions are deleted entirely");
+        assertQuery(session, "SELECT COUNT(*) FROM test_metadata_delete_simple", "SELECT 6");
```

Comment:
We could make these `sum(col2)` to verify that we got the exact rows.

Reply:
Good idea; done.